digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/RoadAddressService.scala

```diff
@@ -158,6 +159,32 @@ class RoadAddressService(viiteClient: SearchViiteClient ) {
     }
   }
 
+
```

Comment:
this method should be move to laneService. Is a specific code for lane

Reply:
laneWithRoadAddress have the same approach as linearAssetWithRoadAddress

Reply:
I think it's like @ricardofmsimoes says, we already have been other cases like linearAssetWithRoadAddress, speedLimitWithRoadAddress, massLimitationWithRoadAddress.

But I think we can try to put a generic method were to be used by all if I looking fine the only difference between most of these methods is the input type (PieceWiseLane or PieceWiseLinearAsset for example), most parts of that types extend from PolyLine class.

Is not possible to create a generic method waiting for PolyLine class and when we pass our types dey execute correctly, please check that.

Reply:
Stayed the same

Reply:
This generic method can't be done because neither PolyLine nor LinearAsset have the properties used on each function this is specific to each sub-type.